### PR TITLE
Discuss

### PR DESCRIPTION
**Pull request by martinthomson**
Lots of minor changes here.  The addition of security considerations is the only substantive change.

### CODE REVIEW

File: draft-ietf-mmusic-sdp-uks.md [hunk 1]
```diff
@@ -336,14 +342,56 @@ Section 3 of {{!TLS13}}:
...
 ~~~
 
-A WebRTC identity assertion is provided as a JSON {{?JSON=RFC8259}} object that
-is encoded into a JSON text.  The resulting string is then encoded using UTF-8
-{{!UTF8=RFC3629}}.  The content of the `external_id_hash` extension are produced
-by hashing the resulting octets with SHA-256 {{!SHA=RFC6234}}.  This produces
-the 32 octets of the `binding_hash` parameter, which is the sole contents of the
+Where an identity assertion has been asserted by a peer, this extension includes
+a SHA-256 hash of the assertion.  An empty value is used to indicates support for
+the extension.
```
Comment (kaduk) [inline draft-ietf-mmusic-sdp-uks.md:347]: maybe "support for the extension, when no local identity assertion is available" (though that would be a bit redundant with the text following the note)

File: draft-ietf-mmusic-sdp-uks.md [hunk 2]
```diff
@@ -336,14 +342,56 @@ Section 3 of {{!TLS13}}:
...
+
+Note:
+
+: For both types of identity assertion, if SHA-256 should prove to be inadequate
+  at some point in the future (see {{?AGILITY=RFC7696}}), a new TLS extension
+  can be defined that uses a different hash function.
+
+Identity bindings might be provided by only one peer.  An endpoint that does not
+produce an identity binding MUST generate an empty `external_id_hash` extension
+in its ClientHello.  An empty extension has a zero-length binding_hash field.
```
Comment (kaduk) [inline draft-ietf-mmusic-sdp-uks.md]: This seems to still assume that any given endpoint will produce a ClientHello

File: draft-ietf-mmusic-sdp-uks.md [hunk 3]
```diff
@@ -336,14 +342,56 @@ Section 3 of {{!TLS13}}:
...
+and MUST cause the receiving endpoint to generate a fatal `decode_error` alert.
+
+In TLS 1.3, the `external_id_hash` extension MUST be sent in the
+EncryptedExtensions message.
+
+
+### Calculating external_id_hash for WebRTC Identity
+
+A WebRTC identity assertion (Section 7 of {{!WEBRTC-SEC}}) is provided as a JSON
+{{!JSON=RFC8259}} object that is encoded into a JSON text.  The JSON test is
```
Comment (kaduk) [inline draft-ietf-mmusic-sdp-uks.md]: s/test/text/

File: draft-ietf-mmusic-sdp-uks.md [hunk 4]
```diff
@@ -336,14 +342,56 @@ Section 3 of {{!TLS13}}:
...
+: For both types of identity assertion, if SHA-256 should prove to be inadequate
+  at some point in the future (see {{?AGILITY=RFC7696}}), a new TLS extension
+  can be defined that uses a different hash function.
+
+Identity bindings might be provided by only one peer.  An endpoint that does not
+produce an identity binding MUST generate an empty `external_id_hash` extension
+in its ClientHello.  An empty extension has a zero-length binding_hash field.
+This allows its peer to include a hash of its identity binding.  An endpoint
+without an identity binding MUST include an empty `external_id_hash` extension
+in its ServerHello or EncryptedExtensions message, to indicate support for the
```
Comment (kaduk) [inline draft-ietf-mmusic-sdp-uks.md]: as does this for the SH/EE.
I don't have any better suggestions than "endpoint acting as TLS {client|server}", though.

File: draft-ietf-mmusic-sdp-uks.md (hunk 1 above)
Comment (martinthomson) [reply draft-ietf-mmusic-sdp-uks.md:347]: Yeah, I don't want to over-rotate on redundancy.

File: draft-ietf-mmusic-sdp-uks.md (hunk 2 above)
Comment (martinthomson) [reply draft-ietf-mmusic-sdp-uks.md]: Ahh, this was intended to be covered in two parts: first the CH bit, then the SH/EE bit.  But changes over time invalidated some of this.  Simplification might help.

**Review by martinthomson — COMMENTED** [review]
Thanks for reviewing Ben.

**Pull request merged**